crates/aptos/src/move_tool/mod.rs

```diff
@@ -853,6 +860,159 @@ impl CliCommand<String> for BuildPublishPayload {
+    #[clap(long)]
+    pub(crate) override_size_check: bool,
+    #[clap(flatten)]
+    pub(crate) included_artifacts_args: IncludedArtifactsArgs,
+    #[clap(flatten)]
+    pub(crate) move_options: MovePackageDir,
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+}
+
+#[async_trait]
+impl CliCommand<TransactionSummary> for CreateObjectAndPublishPackage {
+    fn command_name(&self) -> &'static str {
+        "CreateObjectAndPublishPackage"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<TransactionSummary> {
+        let sender_address = self.txn_options.get_public_key_and_address()?.1;
+        let sequence_number = self.txn_options.sequence_number(sender_address).await? + 1;
+        let object_address = create_object_code_deployment_address(sender_address, sequence_number);
+
+        println!("{}", &self.address_name);
+        self.move_options
+            .add_named_address(self.address_name, object_address.to_string());
+
+        let options = self
+            .included_artifacts_args
+            .included_artifacts
+            .build_options(
+                self.move_options.dev,
+                self.move_options.skip_fetch_latest_git_deps,
+                self.move_options.named_addresses(),
+                self.move_options.bytecode_version,
+                self.move_options.compiler_version,
+                self.move_options.skip_attribute_checks,
+                self.move_options.check_test_code,
+            );
+        let package = BuiltPackage::build(self.move_options.get_package_path()?, options)?;
+        let message = format!(
+            "Do you want to publish this package under the object address {}",
+            object_address
+        );
+        prompt_yes_with_override(&message, self.txn_options.prompt_options)?;
+
+        let payload = aptos_cached_packages::aptos_stdlib::object_code_deployment_publish(
+            bcs::to_bytes(&package.extract_metadata()?).expect("PackageMetadata has BCS"),
+            package.extract_code(),
+        );
+        let size = bcs::serialized_size(&payload)?;
+        println!("package size {} bytes", size);
+
+        if !self.override_size_check && size > MAX_PUBLISH_PACKAGE_SIZE {
+            return Err(CliError::UnexpectedError(format!(
+                "The package is larger than {} bytes ({} bytes)! To lower the size \
+                you may want to include less artifacts via `--included-artifacts`. \
+                You can also override this check with `--override-size-check",
+                MAX_PUBLISH_PACKAGE_SIZE, size
+            )));
+        }
+        self.txn_options
+            .submit_transaction(payload)
+            .await
+            .map(TransactionSummary::from)
+    }
+}
+
+#[derive(Parser)]
+pub struct UpgradeObjectPackage {
+    /// Address of the object containing the package
+    #[clap(long, value_parser = crate::common::types::load_account_arg)]
+    pub(crate) object_address: AccountAddress,
```

Comment:
Should we name this something else instead to make it easier to understand? For example --existing-code. @gregnazario is better with naming here :)

Reply:
Generally, I would keep this value to be the same as the value you used earlier.

`object_address` would be the value you want here since you used it earlier.  I would probably add the documentation:

```
/// Address of the object the package was deployed to
///
/// This must be an already deployed object containing the package
/// if the package is not already created, it will fail.
```

Reply:
Makes sense, the documentation should make this clear.